src/main/java/nl/nn/testtool/storage/file/Reader.java

```diff
@@ -133,7 +132,7 @@ protected List getMetadata(int numberOfRecords, List metadataNames,
 			metadataReadOnly = metadataCacheReadOnly;
 		}
 		List result = new ArrayList();
-		for (int i = 0; i < metadataReadOnly.size() &&  (numberOfRecords == -1 || i < numberOfRecords); i++) {
+		for (int i = 0; i < metadataReadOnly.size() &&  (numberOfRecords == -1 || result.size() < numberOfRecords); i++) {
```

Comment:
When result = 5 and requested numberOfRecords = 10 nothing will be returned?
Maybe it's better to do a separate PR which describes the bug you are trying to fix?